src/awkward/operations/ak_almost_equal.py

```diff
@@ -48,6 +51,12 @@ def almost_equal(
     right = to_layout(right, allow_record=False).to_packed()
 
     backend = backend_of(left, right)
+    if not backend.nplike.known_data:
+        raise wrap_error(
+            TypeError(
+                "Awkward Arrays with typetracer backends cannot be compared with `ak.almost_equal`."
+            )
+        )
```

Comment:
We would not be able to produce very meaningful results here. We could return a 0-D scalar for not-known, but I chose just to make this an error. In general, we already know that boolean-like operations on arrays are not likely to be well supported in typetracer due to the need for concrete values.

That's why we have a proposal for `empty_if_typetracer`.

Not totally convinced here, though - I could be swayed.

Reply:
To add - if we support `known_data = False` here, then we need additional flags for what should happen if `left` has a known length but `right` has `unknown_length`. I think the right answer is probably to add typetracer support here, but I'm leaving this as a discussion point.

Reply:
The only meaningful thing that `ak.almost_equal` could do for typetracers would be to verify that they have the same high-level types.

If that short-circuits at the beginning of the function (if either array is typetracer, only check types), that would be another way of handling it, but then this error would become dead code.

It's also fine to raise errors if one is a typetracer array, as this one has it. That also allows us to put off a decision, since an error can be replaced by non-error behavior.

Reply:
Well, typetracer can also check buffer lengths; in some cases these are known.